tools/release

```diff
@@ -0,0 +1,124 @@
+[ "$commit_msg" == "Release Zulip Server $version." ] \
+    || fail "Expected commit message: Release Zulip Server $version."
+
+# Provision
+echo "Provisioning.."
+if ! ./tools/provision >/dev/null; then
+    cat var/log/provision.log
+    fail "Provisioning failed!"
+fi
+
+# Check lint passes for the changelog
+./tools/lint --skip gitlint docs/overview/changelog.md \
+    || fail "Changelog does not pass lint"
+
+# Check the date is correct for the release
+release_line=$(grep -x -E -m 1 -o "### $version -- ([0-9-]+)" docs/overview/changelog.md) \
+    || fail "docs/overview/changelog.md does not contain a line for $version"
+[ "$release_line" == "### $version -- $(TZ=America/Los_Angeles date +%F)" ] \
+    || fail "Date in docs/overview/changelog.md does not match $(date +%F)"
+
+extract_version() {
+    setting="$1"
+    value=$(SETTING="$setting" perl -nle 'print $2 if /$ENV{SETTING} = (\"?)([^"]+)\1$/' version.py | head -n1)
+    [ -n "$value" ] || fail "Could not find $setting in version.py"
+    echo "$value"
+}
+
+# Check ZULIP_VERSION and LATEST_RELEASE_VERSION are set appropriately
+[ "$(extract_version "ZULIP_VERSION")" == "$version" ] \
+    || fail "ZULIP_VERSION in version.py does not match $version"
+[ "$(extract_version "LATEST_RELEASE_VERSION")" == "$version" ] \
+    || fail "LATEST_RELEASE_VERSION in version.py does not match $version"
+
+if [ -n "$is_major_release" ]; then
+    [ "$(extract_version "LATEST_MAJOR_VERSION")" == "$version" ] \
+        || fail "LATEST_MAJOR_VERSION in version.py does not match $version"
+
+    # Check that there is an API version bump, which is documented
+    changed_api_level=$(git diff-tree -G API_FEATURE_LEVEL HEAD -- version.py)
+    [ -n "$changed_api_level" ] || fail "$version did not adjust API_FEATURE_LEVEL in version.py"
+    feature_level="$(extract_version "API_FEATURE_LEVEL")"
+    grep -q -F -x "**Feature level $feature_level**" templates/zerver/api/changelog.md \
+        || fail "Feature level $feature_level is not documented in templates/zerver/api/changelog.md"
+fi
+
+# Check we are auth'd to Github, so we can upload the release
+type gh >/dev/null 2>&1 \
+    || fail "The 'gh' CLI tool is not installed; see https://cli.github.com/"
+gh auth status \
+    || fail "Not authenticated to github"
+
+# Extract the changelog, print it
+changelog=$(VERSION="$version" perl -nle '$v=quotemeta($ENV{VERSION}); print if $rc = /^### $v/ .. /^#{1,3} (?!$v)/ and $rc !~ /E0/' docs/overview/changelog.md)
+echo "$changelog"
+
+echo git tag "$version"
+
+# Commits need to be pushed to the remote so that "branched from" is correct
+remote="$(git config zulip.zulipRemote)" || remote=upstream
+echo git push "$remote" "$branch:$branch"
+
+OUTDIR=$(mktemp -d)
+trap 'rm -r "$OUTDIR"' EXIT
+export OUTDIR
+
+./tools/build-release-tarball "$version"
+
+TARBALL="$OUTDIR/zulip-server-$version.tar.gz"
+if ! [ -f "$TARBALL" ]; then
+    echo "Did not find expected $TARBALL!"
+    exit 1
+fi
+
```

Comment:
Should we add a pause point here when you can hit enter to proceed with uploading or `ctrl+C` to cancel? The type of failure mode I'm imagining is you notice the changelog didn't format properly, or something.